### PR TITLE
Add ProcessorPackageFileService to allow processors to store their .nupkgs

### PR DESCRIPTION
**Pull request by joelverhagen**
Progress on https://github.com/NuGet/Engineering/issues/1207

1. Add ProcessorPackageFileService to allow processors to store their .nupkgs
1. Update ServerCommon
1. Update NuGetGallery.Core
1. Fix some `PackageDownloadTimeout` oopsies

This is a PR preparing for SignatureProcessor stripping repository signatures.

### CODE REVIEW

File: src/Validation.Common.Job/FileStreamUtility.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+using System.IO;
+
+namespace NuGet.Jobs.Validation
+{
+    public static class FileStreamUtility
+    {
+        public const int BufferSize = 8192;
+
+        public static FileStream GetTemporaryFile()
```
Comment (loic-sharma) [inline src/Validation.Common.Job/FileStreamUtility.cs]: >GetTemporaryFile [](start = 33, length = 16)

Why is this separation from the PackageDownloader necessary?

Comment (joelverhagen) [reply src/Validation.Common.Job/FileStreamUtility.cs]: Used in two places, later. In particular, when we strip repository signatures we need to write the new packages to a new stream.

File: src/Validation.Common.Job/Storage/ProcessorPackageFileService.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            packageFile.Position = 0;
+
+            return _fileStorageService.SaveFileAsync(
+                CoreConstants.ValidationFolderName,
+                fileName,
+                packageFile,
+                overwrite: true);
+        }
+
+        private string BuildFileName(string packageId, string packageNormalizedVersion, Guid validationId)
```
Comment (loic-sharma) [inline src/Validation.Common.Job/Storage/ProcessorPackageFileService.cs]: >BuildFileName [](start = 23, length = 13)

Minor suggestion: `GetReadAndDeleteUriAsync` and `SaveAsync` are public methods. Maybe this method should do null checking for `packageId` and `packageNormalizedVersion`?

Comment (joelverhagen) [reply src/Validation.Common.Job/Storage/ProcessorPackageFileService.cs]: Sure, sounds fine.

File: tests/Validation.Common.Job.Tests/FileStreamUtilityFacts.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                {
+                    fileName = stream.Name;
+                    writer.WriteLine(FileContent);
+                    writer.Flush();
+
+                    // Act & Assert
+                    Assert.True(File.Exists(fileName), "The file should still exist.");
+                }
+
+                Assert.False(File.Exists(fileName), "The file should still exist.");
```
Comment (loic-sharma) [inline tests/Validation.Common.Job.Tests/FileStreamUtilityFacts.cs]: >The file should still exist [](start = 53, length = 27)

"should still" -> "should not exist"

Comment (joelverhagen) [reply tests/Validation.Common.Job.Tests/FileStreamUtilityFacts.cs]: Fixed.

File: src/Validation.Common.Job/Storage/ProcessorPackageFileService.cs (hunk 2 above)
Comment (joelverhagen) [reply src/Validation.Common.Job/Storage/ProcessorPackageFileService.cs]: Fixed.

**Pull request merged**